### ISSUE TITLE
When the window size changes, the pop-up layer position is incorrect

### ISSUE THREAD

**Issue by srnrider53** [AhmedLSayed9/dropdown_button2#243]
![image](https://github.com/AhmedLSayed9/dropdown_button2/assets/28307712/e71a01fa-4168-4b62-b4c6-e8ce8ca68887)

**Comment by AndryCU** [comment]
This is a minimal example where you can reproduce this issue:

```
import 'package:dropdown_button2/dropdown_button2.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const DropdownMenuExample());
}

class DropdownMenuExample extends StatelessWidget {
  const DropdownMenuExample({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        useMaterial3: true,
        colorSchemeSeed: Colors.green,
      ),
      home: Scaffold(
        body: Center(
          child: Padding(
            padding: const EdgeInsets.symmetric(vertical: 20),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: <Widget>[
                DropdownButton2<int>(
                  buttonStyleData: const ButtonStyleData(
                    padding: EdgeInsets.zero,
                  ),
                  onChanged: (value) {},
                  items: [
                    for (int i = 1; i <= 5; i++)
                      DropdownMenuItem<int>(
                        value: i,
                        child: Text(i.toString()),
                      ),
                  ],
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

And this is the current behavior:

[Screencast from 25-03-24 21:21:13.webm](https://github.com/AhmedLSayed9/dropdown_button2/assets/74194123/8e3d5d8d-abf3-4d1f-8e69-e6c9119f92ae)

Proposal: **Closing the menu immediately changes the screen size.** 

This is how the Flutter DropdownMenu behaves:

[Screencast from 25-03-24 21:38:44.webm](https://github.com/AhmedLSayed9/dropdown_button2/assets/74194123/6aa7b7a9-2f7b-4a04-85b1-9c3d8afd9b89)

**Comment by srnrider53** [comment]
> This is a minimal example where you can reproduce this issue:
> 
> ```
> import 'package:dropdown_button2/dropdown_button2.dart';
> import 'package:flutter/material.dart';
> 
> void main() {
>   runApp(const DropdownMenuExample());
> }
> 
> class DropdownMenuExample extends StatelessWidget {
>   const DropdownMenuExample({super.key});
> 
>   @override
>   Widget build(BuildContext context) {
>     return MaterialApp(
>       theme: ThemeData(
>         useMaterial3: true,
>         colorSchemeSeed: Colors.green,
>       ),
>       home: Scaffold(
>         body: Center(
>           child: Padding(
>             padding: const EdgeInsets.symmetric(vertical: 20),
>             child: Row(
>               mainAxisAlignment: MainAxisAlignment.center,
>               children: <Widget>[
>                 DropdownButton2<int>(
>                   buttonStyleData: const ButtonStyleData(
>                     padding: EdgeInsets.zero,
>                   ),
>                   onChanged: (value) {},
>                   items: [
>                     for (int i = 1; i <= 5; i++)
>                       DropdownMenuItem<int>(
>                         value: i,
>                         child: Text(i.toString()),
>                       ),
>                   ],
>                 ),
>               ],
>             ),
>           ),
>         ),
>       ),
>     );
>   }
> }
> ```
> 
> And this is the current behavior:
> 
>  Screencast.from.25-03-24.21.21.13.webm 
> Proposal: **Closing the menu immediately changes the screen size.**
> 
> This is how the Flutter DropdownMenu behaves:
> 
>  Screencast.from.25-03-24.21.38.44.webm

ok. thx!

**Comment by adonisRodxander** [comment]
Same issue

**Comment by ankitgitac** [comment]
@AhmedLSayed9 Is this bug getting fixed?

**Comment by adonisRodxander** [comment]
Any update?

**Comment by AhmedLSayed9** [comment]
This has been fixed at `v3.0.0-beta.16`

**Comment by Suraj-Smartcache** [comment]
@AhmedLSayed9 its working perfectly in `3.0.0-beta.23`. Thanks!
